hummingbot/core/cpp/OrderBookEntry.cpp

```diff
@@ -60,16 +60,16 @@ void truncateOverlapEntriesDex(std::set<OrderBookEntry> &bidBook, std::set<Order
 void truncateOverlapEntriesCentralised(std::set<OrderBookEntry> &bidBook, std::set<OrderBookEntry> &askBook) {
     std::set<OrderBookEntry>::reverse_iterator bidIterator = bidBook.rbegin();
     std::set<OrderBookEntry>::iterator askIterator = askBook.begin();
-    while (bidIterator != bidBook.rend() && askIterator != askBook.end()) {
+    while (bidIterator != bidBook.rend() && askIterator != askBook.end() && !bidBook.empty() && !askBook.empty()) {
         const OrderBookEntry& topBid = *bidIterator;
         const OrderBookEntry& topAsk = *askIterator;
         if (topBid.price >= topAsk.price) {
             if (topBid.updateId > topAsk.updateId) {
                 askBook.erase(askIterator++);
             } else {
                 std::set<OrderBookEntry>::iterator eraseIterator = (std::next(bidIterator)).base();
-                bidIterator++;
                 bidBook.erase(eraseIterator);
+                bidIterator++;
```

Comment:
Moving the `bidIterator++` after the erase looks wrong to me... basically it means you're advancing an erased iterator - and it should cause even more segfaults than before. Are you sure this is correct?